extensions/arc/src/ui/dialogs/addPGExtensionsDialog.ts

```diff
@@ -0,0 +1,88 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { Deferred } from '../../common/promise';
+import * as loc from '../../localizedConstants';
+import { cssStyles } from '../../constants';
+import { InitializingComponent } from '../components/initializingComponent';
+import { PostgresModel } from '../../models/postgresModel';
+
+export class AddPGExtensionsDialog extends InitializingComponent {
+	protected modelBuilder!: azdata.ModelBuilder;
+
+	protected extensionsListInputBox!: azdata.InputBoxComponent;
+
+	protected _completionPromise = new Deferred<string | undefined>();
+
+	constructor(protected _model: PostgresModel) {
+		super();
+	}
+
+	public showDialog(dialogTitle: string): azdata.window.Dialog {
+		const dialog = azdata.window.createModelViewDialog(dialogTitle);
+		dialog.cancelButton.onClick(() => this.handleCancel());
+		dialog.registerContent(async view => {
+			this.modelBuilder = view.modelBuilder;
+
+			const info = this.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+				value: loc.extensionsFunction,
+				CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+			}).component();
+
+			const link = this.modelBuilder.hyperlink().withProperties<azdata.HyperlinkComponentProperties>({
+				label: loc.extensionsLearnMore,
+				url: 'https://docs.microsoft.com/azure/azure-arc/data/using-extensions-in-postgresql-hyperscale-server-group',
+			}).component();
+
+			const infoAndLink = this.modelBuilder.flexContainer().withLayout({ flexWrap: 'wrap' }).component();
+			infoAndLink.addItem(info, { CSSStyles: { 'margin-right': '5px' } });
+			infoAndLink.addItem(link);
+
+			this.extensionsListInputBox = this.modelBuilder.inputBox()
+				.withProperties<azdata.InputBoxProperties>({
```

Comment:
FYI if you use `withProps` that gets you the proper typing for free without having to specify the properties type yourself.